### PR TITLE
snappy: bump to snappy/1.1.9

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **snappy/1.1.9**

bump

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/snappy/all/patches/1.1.9-fix-inlining-failure-functino-body-can-be-overwritten-at-link-time.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
++++ snappy-stubs-internal.h
+@@ -100,7 +100,7 @@
+ 
+ // Inlining hints.
+ #ifdef HAVE_ATTRIBUTE_ALWAYS_INLINE
+-#define SNAPPY_ATTRIBUTE_ALWAYS_INLINE __attribute__((always_inline))
++#define SNAPPY_ATTRIBUTE_ALWAYS_INLINE __attribute__((inline))
+ #else
+ #define SNAPPY_ATTRIBUTE_ALWAYS_INLINE
+ #endif
```
Comment (SpaceIm) [inline recipes/snappy/all/patches/1.1.9-fix-inlining-failure-functino-body-can-be-overwritten-at-link-time.patch:11]: unother possible fix:

```patch
--- a/snappy.cc
+++ b/snappy.cc
@@ -1014,7 +1014,7 @@ void MemMove(ptrdiff_t dst, const void* src, size_t size) {
 }
 
 SNAPPY_ATTRIBUTE_ALWAYS_INLINE
-size_t AdvanceToNextTag(const uint8_t** ip_p, size_t* tag) {
+inline size_t AdvanceToNextTag(const uint8_t** ip_p, size_t* tag) {
   const uint8_t*& ip = *ip_p;
   // This section is crucial for the throughput of the decompression loop.
   // The latency of an iteration is fundamentally constrained by the
```

**Comment by SpaceIm** [comment]
I've already tested snappy 1.1.9 and there is another issue to fix for clang < 9 and apple-clang < 11: https://github.com/SpaceIm/conan-snappy/actions/runs/845643070

```
[ 20%] Building CXX object source_subfolder/CMakeFiles/snappy.dir/snappy-c.cc.o
[ 40%] Building CXX object source_subfolder/CMakeFiles/snappy.dir/snappy-sinksource.cc.o
[ 60%] Building CXX object source_subfolder/CMakeFiles/snappy.dir/snappy-stubs-internal.cc.o
[ 80%] Building CXX object source_subfolder/CMakeFiles/snappy.dir/snappy.cc.o
/home/conan/.conan/data/snappy/1.1.9/bincrafters/testing/build/1a1ab1b5ec185a3fe2d9c8e7c5b43dde10fa236a/source_subfolder/snappy.cc:1033:36: error: invalid output constraint '=@ccz' in asm
      : [tag_type] "+r"(tag_type), "=@ccz"(is_literal));
                                   ^
1 error generated.
```

File: recipes/snappy/all/conandata.yml [hunk 2]
```diff
@@ -1,7 +1,14 @@
...
+    url: "https://github.com/google/snappy/archive/1.1.8.tar.gz"
+    sha256: "16b677f07832a612b0836178db7f374e414f94657c138e6993cbfc5dcc58651f"
   "1.1.7":
-    url: https://github.com/google/snappy/archive/1.1.7.tar.gz
-    sha256: 3dfa02e873ff51a11ee02b9ca391807f0c8ea0529a4924afa645fbf97163f9d4
+    url: "https://github.com/google/snappy/archive/1.1.7.tar.gz"
+    sha256: "3dfa02e873ff51a11ee02b9ca391807f0c8ea0529a4924afa645fbf97163f9d4"
+patches:
+  "1.1.9":
+    - patch_file: "patches/1.1.9-fix-inlining-failure-functino-body-can-be-overwritten-at-link-time.patch"
```
Comment (SpaceIm) [inline recipes/snappy/all/conandata.yml:13]: Wow, maybe reduce size of filename? 😄

**Comment by madebr** [comment]
> I've already tested snappy 1.1.9 and there is another issue to fix for clang < 9 and apple-clang < 11: https://github.com/SpaceIm/conan-snappy/actions/runs/845643070
> 
> ```
> [ 20%] Building CXX object source_subfolder/CMakeFiles/snappy.dir/snappy-c.cc.o
> [ 40%] Building CXX object source_subfolder/CMakeFiles/snappy.dir/snappy-sinksource.cc.o
> [ 60%] Building CXX object source_subfolder/CMakeFiles/snappy.dir/snappy-stubs-internal.cc.o
> [ 80%] Building CXX object source_subfolder/CMakeFiles/snappy.dir/snappy.cc.o
> /home/conan/.conan/data/snappy/1.1.9/bincrafters/testing/build/1a1ab1b5ec185a3fe2d9c8e7c5b43dde10fa236a/source_subfolder/snappy.cc:1033:36: error: invalid output constraint '=@ccz' in asm
>       : [tag_type] "+r"(tag_type), "=@ccz"(is_literal));
>                                    ^
> 1 error generated.
> ```

Thanks for the heads-up.
These lines are causing a problem:
https://github.com/google/snappy/blob/2b63814b15a2aaae54b7943f0cd935892fae628f/snappy.cc#L1029-L1033

Weird how `__GNUC__` is defined when compiling with llvm.

Perhaps that block needs an extra check:
```patch
-#if defined(__GNUC__) && defined(__x86_64__)
+#if defined(__GNUC__) && defined(__x86_64__) && !defined(__clang__)
```

File: recipes/snappy/all/conandata.yml (hunk 2 above)
Comment (madebr) [reply recipes/snappy/all/conandata.yml]: Hey, I don't work with DOS anymore :rofl:
Is `fxinline.dif` ok? :stuck_out_tongue:

**Comment by SpaceIm** [comment]
Suggestion to remove Werror and provide finer grained CMake options.

```python
    def _patch_sources(self):
        for patch in self.conan_data.get("patches", {}).get(self.version, []):
            tools.patch(**patch)
        if tools.Version(self.version) >= "1.1.9":
            # No warnings as errors
            tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
                                  "set(CMAKE_CXX_FLAGS \"${CMAKE_CXX_FLAGS} -Werror\")",
                                  "")

    def _configure_cmake(self):
        if self._cmake:
            return self._cmake
        self._cmake = CMake(self)
        self._cmake.definitions["SNAPPY_BUILD_TESTS"] = False
        if tools.Version(self.version) >= "1.1.9":
            self._cmake.definitions["SNAPPY_BUILD_BENCHMARKS"] = False
        if tools.Version(self.version) >= "1.1.8":
            self._cmake.definitions["SNAPPY_FUZZING_BUILD"] = False
            self._cmake.definitions["SNAPPY_REQUIRE_AVX"] = False
            self._cmake.definitions["SNAPPY_REQUIRE_AVX2"] = False
            self._cmake.definitions["SNAPPY_INSTALL"] = True
        self._cmake.configure(build_folder=self._build_subfolder)
        return self._cmake

    def build(self):
        self._patch_sources()
        cmake = self._configure_cmake()
        cmake.build()
```

**Comment by madebr** [comment]
The error the `always_inline` patch is fixing is emitted by gcc, which does not receive the `-Werror` flag.
Only clang gets `-Werror`.
https://github.com/google/snappy/blob/2b63814b15a2aaae54b7943f0cd935892fae628f/CMakeLists.txt#L64-L74

I will probably have to apply your suggestion though.

**Comment by madebr** [comment]
@SpaceIm

The `__GNUC__` macro means that the compiler supports the GNU C extensions.
See https://gcc.gnu.org/onlinedocs/cpp/Common-Predefined-Macros.html#Common-Predefined-Macros

So you cannot use it to differentiate between gcc and llvm.

File: recipes/snappy/all/conanfile.py [hunk 3]
```diff
@@ -36,19 +45,28 @@ def configure(self):
...
-        extracted_dir = "{}-{}".format(self.name, self.version)
-        os.rename(extracted_dir, self._source_subfolder)
+        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
 
     def _configure_cmake(self):
         if self._cmake:
             return self._cmake
         self._cmake = CMake(self)
+        self._cmake.verbose=True
+        self._cmake.verbose = True
```
Comment (madebr) [inline recipes/snappy/all/conanfile.py:55]:
```suggestion
```

File: recipes/snappy/all/patches/1.1.9-0003-fix-clobber-list-older-llvm.patch [hunk 4]
```diff
@@ -0,0 +1,12 @@
+asm clobbers do not work for clang < 9 and apple-clang < 11 (found by SpaceIm)
+--- snappy.cc
++++ snappy.cc
+@@ -1026,7 +1026,7 @@
+   size_t literal_len = *tag >> 2;
+   size_t tag_type = *tag;
+   bool is_literal;
+-#if defined(__GNUC__) && defined(__x86_64__)
++#if defined(__GNUC__) && defined(__x86_64__) && ( (!defined(__clang__) && !defined(__APPLE__)) || (!defined(__APPLE__) && defined(__clang__) && (__clang_major__ > 9)) || (defined(__APPLE__) && (__clang_major__ > 11)) )
```
Comment (madebr) [inline recipes/snappy/all/patches/1.1.9-0003-fix-clobber-list-older-llvm.patch:9]:
```suggestion
+#if defined(__GNUC__) && defined(__x86_64__) && ( (!defined(__clang__) && !defined(__APPLE__)) || (!defined(__APPLE__) && defined(__clang__) && (__clang_major__ > 9)) || (defined(__APPLE__) && defined(__clang__) && (__clang_major__ > 11)) )
```

**Review by SSE4 — APPROVED** [review]

File: recipes/snappy/all/patches/1.1.9-0003-fix-clobber-list-older-llvm.patch [hunk 5]
```diff
@@ -0,0 +1,12 @@
+asm clobbers do not work for clang < 9 and apple-clang < 11 (found by SpaceIm)
+--- snappy.cc
++++ snappy.cc
+@@ -1026,7 +1026,7 @@
+   size_t literal_len = *tag >> 2;
+   size_t tag_type = *tag;
+   bool is_literal;
+-#if defined(__GNUC__) && defined(__x86_64__)
++#if defined(__GNUC__) && defined(__x86_64__) && ( (!defined(__clang__) && !defined(__APPLE__)) || (!defined(__APPLE__) && defined(__clang__) && (__clang_major__ > 9)) || (defined(__APPLE__) && defined(__clang__) && (__clang_major__ > 11)) )
```
Comment (SpaceIm) [inline recipes/snappy/all/patches/1.1.9-0003-fix-clobber-list-older-llvm.patch:9]:
```suggestion
+#if defined(__GNUC__) && defined(__x86_64__) && ( (!defined(__clang__) && !defined(__APPLE__)) || (!defined(__APPLE__) && defined(__clang__) && (__clang_major__ >= 9)) || (defined(__APPLE__) && defined(__clang__) && (__clang_major__ >= 11)) )
```

It should work with clang 9 and apple-clang 11

File: recipes/snappy/all/patches/1.1.9-0003-fix-clobber-list-older-llvm.patch [hunk 6]
```diff
@@ -0,0 +1,12 @@
+asm clobbers do not work for clang < 9 and apple-clang < 11 (found by SpaceIm)
+--- snappy.cc
++++ snappy.cc
+@@ -1026,7 +1026,7 @@
+   size_t literal_len = *tag >> 2;
+   size_t tag_type = *tag;
+   bool is_literal;
+-#if defined(__GNUC__) && defined(__x86_64__)
++#if defined(__GNUC__) && defined(__x86_64__) && ( (!defined(__clang__) && !defined(__APPLE__)) || (!defined(__APPLE__) && defined(__clang__) && (__clang_major__ >= 9)) || (defined(__APPLE__) && defined(__clang__) && (__clang_major__ >= 11)) )
```
Comment (madebr) [inline recipes/snappy/all/patches/1.1.9-0003-fix-clobber-list-older-llvm.patch:9]:
```suggestion
+#if defined(__GNUC__) && defined(__x86_64__) && ( (!defined(__clang__) && !defined(__APPLE__)) || (!defined(__APPLE__) && defined(__clang__) && (__clang_major__ >= 9)) || (defined(__APPLE__) && defined(__clang__) && (__clang_major__ > 11)) )
```

**Review by SpaceIm — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**